src/org/rust/lang/structure/RustStructTreeElement.kt

```diff
@@ -0,0 +1,24 @@
+package org.rust.lang.structure
+
+import com.intellij.ide.structureView.StructureViewTreeElement
+import com.intellij.ide.structureView.impl.common.PsiTreeElementBase
+import org.rust.lang.core.psi.RustStructDeclField
+import org.rust.lang.core.psi.RustStructItem
+
+class RustStructTreeElement(element: RustStructItem) : PsiTreeElementBase<RustStructItem>(element) {
+
+    override fun getPresentableText(): String? = element?.name
+
+    override fun getChildrenBase(): Collection<StructureViewTreeElement> {
+        val result = arrayListOf<StructureViewTreeElement>()
+
+        for (field in getDeclFields().orEmpty())
```

Comment:
You can use `getDeclFields().orEmpty().map { RustStructDeclFieldTreeElement(it) }`